File: RUDRA/motor_control.cpp
#include <iostream>
#include <thread> // For std::this_thread::sleep_for
#include <chrono> // For std::chrono::seconds

// Function to print the binary representation of a number
void printBinary(int number) {
    for (int i = 7; i >= 0; --i) {
        std::cout << ((number >> i) & 1); // Shift the bits and mask with 1
    }
    std::cout << std::endl;
}

int main() {
    // Move motor 1 forward at half speed
    printBinary(127); // Motor 1 forward (half speed)
    std::this_thread::sleep_for(std::chrono::seconds(2)); // Wait for 2 seconds

    // Stop the motor
    printBinary(0); // Stop motor 1
    std::this_thread::sleep_for(std::chrono::seconds(2)); // Wait for 2 seconds

    // Move motor 1 in reverse at half speed
    printBinary(192); // Motor 1 reverse (half speed)
    std::this_thread::sleep_for(std::chrono::seconds(2)); // Wait for 2 seconds

    // Stop the motor again
    printBinary(0); // Stop motor 1
    std::this_thread::sleep_for(std::chrono::seconds(2)); // Wait for 2 seconds

    return 0;
}

File: RUDRA/motor_control.py
void setup() {
  Serial.begin(9600);  // Initialize serial communication
}

void loop() {
  // Move motor 1 forward at half speed
  Serial.println(127); // Motor 1 forward (half speed)
  delay(2000); // Wait for 2 seconds

  // Stop the motor
  Serial.println(0); // Stop motor 1
  delay(2000); // Wait for 2 seconds

  // Move motor 1 in reverse at half speed
  Serial.println(192); // Motor 1 reverse (half speed)
  delay(2000); // Wait for 2 seconds

  // Stop the motor again
  Serial.println(0); // Stop motor 1
  delay(2000); // Wait for 2 seconds
}

